### ISSUE TITLE
Git PlainClone is throwing reference delta not found error

### ISSUE THREAD

**Issue by ajaykn** [go-git/go-git#495]
I am trying to clone repo: 
https://github.com/OctopusDeploy/install-octopus-cli-action

And seeing the error `reference delta not found`

## git.PlainClone is failing

```go
r, err := git.PlainClone("temp/"+repo_nwo, true, &git.CloneOptions{
    SingleBranch: false,
    URL:          "https://github.com/" + repo_nwo,
    Progress:     os.Stdout,
})
```
Output:

```log
Enumerating objects: 8163, done.
Counting objects: 100% (1465/1465), done.
Compressing objects: 100% (854/854), done.
Total 8163 (delta 811), reused 964 (delta 515), pack-reused 6698
reference delta not found
```

## git.Clone is working fine

```go
storage := memory.NewStorage()
r, err := git.Clone(storage, nil, &git.CloneOptions{
    SingleBranch: false,
    URL:          "https://github.com/" + repo_nwo,
    Progress:     os.Stdout,
})
```

Output:
```log
Enumerating objects: 8163, done.
Counting objects: 100% (1465/1465), done.
Compressing objects: 100% (856/856), done.
Total 8163 (delta 811), reused 962 (delta 513), pack-reused 6698
```

**Comment by ZauberNerd** [comment]
I can reproduce your issue with the latest master, my PR (#485 ) fixes it:
```golang
package main

import (
  "os"
  "github.com/go-git/go-git/v5"
)

func main() {
  _, err := git.PlainClone("temp/octopus", true, &git.CloneOptions{
    SingleBranch: false,
    URL:          "https://github.com/OctopusDeploy/install-octopus-cli-action",
    Progress:     os.Stdout,
  })

  if err != nil {
    println(err.Error())
  }
}
```

* Running the above code via `go run main.go` fails
* Using latest [`master`](https://github.com/go-git/go-git/commits/master) via `go mod edit -replace github.com/go-git/go-git/v5=github.com/go-git/go-git/v5@c785af3f4559ebac52c42f12d17cb118aac383ad && go mod tidy` (to check if #392 fixes it) and  then running `go run main.go` fails
* Using my PR (#485 ) via `go mod edit -replace github.com/go-git/go-git/v5=github.com/ZauberNerd/go-git/v5@29ec1bc1e5db2c38fea941603da32070e0af5a91 && go mod tidy` and  then running `go run main.go` works

**Comment by kaiohenricunha** [comment]
I ran into the same issue and @ZauberNerd 's [PR](https://github.com/go-git/go-git/pull/485) solved it.

Thanks!

But, why is it taking so long for PRs to be merged?

**Comment by kyparisisg** [comment]
Hey folks, I am running to the same issue here!
Any thoughts when the [PR](https://github.com/go-git/go-git/pull/485) will get merged?
Thank you!

**Comment by AriehSchneier** [comment]
Fixed by: #485

**Comment by AriehSchneier** [comment]
@pjbgf - it looks like this should now be closed as fixed by #485

**Comment by pjbgf** [comment]
Nice one @AriehSchneier. Closing as per #485.
